modules/images/image-loading-optimization/storage/rest-api.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * REST API integration for the module.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+const ILO_REST_API_NAMESPACE = 'image-loading-optimization/v1';
+
+const ILO_PAGE_METRIC_STORAGE_ROUTE = '/image-loading-optimization/page-metric-storage';
+
+/**
+ * Register endpoint for storage of page metric.
+ */
+function ilo_register_endpoint() {
+
+	$dom_rect_schema = array(
+		'type'       => 'object',
+		'properties' => array(
+			'width'  => array(
+				'type'    => 'number',
+				'minimum' => 0,
+			),
+			'height' => array(
+				'type'    => 'number',
+				'minimum' => 0,
+			),
+			// TODO: There are other properties to define if we need them: x, y, top, right, bottom, left.
+		),
+	);
+
+	register_rest_route(
+		ILO_REST_API_NAMESPACE,
+		ILO_PAGE_METRIC_STORAGE_ROUTE,
+		array(
+			'methods'             => 'POST',
+			'callback'            => static function ( WP_REST_Request $request ) {
+				return ilo_handle_rest_request( $request );
+			},
```

Comment:
Nit-pick:
```suggestion
			'callback'            => 'ilo_handle_rest_request',
```

Reply:
Actually I did it this way intentionally for better static analysis.

Reply:
Namely, the `ilo_handle_rest_request` function otherwise is not detected as being used if it is just passed as a string. In PHP 8.1 this can be avoided with [first class callable syntax](https://www.php.net/manual/en/functions.first_class_callable_syntax.php) like so:

```suggestion
			'callback'            => ilo_handle_rest_request(...),
```

Reply:
@westonruter Alternatively you could just add `@uses ilo_handle_rest_request` to the `ilo_register_endpoint` docblock

Reply:
@swissspidy Good point. Although that seems more error prone as it doesn't specify _how_ the function is used including what parameters are passed to it.

Reply:
Right. PHPStan would help with detailed type checks.
The annotation just addresses the concern about it being marked as unused.